pinot-common/src/main/java/com/linkedin/pinot/common/utils/EqualityUtils.java

```diff
@@ -40,11 +39,11 @@ public static boolean isEqual(long left, long right) {
   }
 
   public static boolean isEqual(float left, float right) {
-    return Float.compare(left, right) == 0;
+    return left == right;
   }
 
   public static boolean isEqual(double left, double right) {
-    return Double.compare(left, right) == 0;
+    return left == right;
```

Comment:
the method call might get optimized but having function for built-in operator is odd